src/libraries/System.Private.CoreLib/src/System/Boolean.cs

```diff
@@ -187,6 +187,42 @@ public int CompareTo(bool value)
         // Static Methods
         //
 
+        // Custom string compares for early application use by config switches, etc
+        //
+#if MONO
+        // We have to keep these implementations for Mono here because MemoryExtensions.Equals("True", OrdinalIgnoreCase)
+        // triggers CompareInfo static initialization which is not desired when we parse configs on start.
+        // TODO: Remove once Mono aligns its behavior with CoreCLR around .beforefieldinit
+        // https://github.com/dotnet/runtime/issues/77513
+        internal static bool IsTrueStringIgnoreCase(ReadOnlySpan<char> value)
+        {
+            // "true" as a ulong, each char |'d with 0x0020 for case-insensitivity
+            ulong true_val = BitConverter.IsLittleEndian ? 0x65007500720074ul : 0x74007200750065ul;
+            return value.Length == 4 &&
+                   (MemoryMarshal.Read<ulong>(MemoryMarshal.AsBytes(value)) | 0x0020002000200020) == true_val;
+        }
+
+        internal static bool IsFalseStringIgnoreCase(ReadOnlySpan<char> value)
+        {
+            // "fals" as a ulong, each char |'d with 0x0020 for case-insensitivity
+            ulong fals_val = BitConverter.IsLittleEndian ? 0x73006C00610066ul : 0x660061006C0073ul;
+            return value.Length == 5 &&
+                   (((MemoryMarshal.Read<ulong>(MemoryMarshal.AsBytes(value)) | 0x0020002000200020) == fals_val) &
+                    ((value[4] | 0x20) == 'e'));
```

Comment:
Shouldn't the big ensian constant end in 00?

Reply:
Ah, no :-)

Reply:
@jakobbotsch the BE version of "fals" doesn't end with 00 either and it works so I assume it's correct? 🤷 context: https://github.com/dotnet/runtime/pull/65078